entities/src/main/java/com/flipkart/varadhi/entities/auth/ResourceAction.java

```diff
@@ -28,15 +28,26 @@ public enum ResourceAction {
     TOPIC_GET(ResourceType.TOPIC, "get"),
     TOPIC_DELETE(ResourceType.TOPIC, "delete"),
     TOPIC_UPDATE(ResourceType.TOPIC, "update"),
+    TOPIC_LIST(ResourceType.TOPIC, "list"),
     TOPIC_CONSUME(ResourceType.TOPIC, "consume"),
     TOPIC_PRODUCE(ResourceType.TOPIC, "produce"),
+    TOPIC_IAM_POLICY_SET(ResourceType.TOPIC, "set_iam_policy"),
+    TOPIC_IAM_POLICY_GET(ResourceType.TOPIC, "get_iam_policy"),
+    TOPIC_IAM_POLICY_DELETE(ResourceType.TOPIC, "delete_iam_policy"),
 
     SUBSCRIPTION_CREATE(ResourceType.SUBSCRIPTION, "create"),
     SUBSCRIPTION_GET(ResourceType.SUBSCRIPTION, "get"),
     SUBSCRIPTION_LIST(ResourceType.SUBSCRIPTION, "list"),
     SUBSCRIPTION_DELETE(ResourceType.SUBSCRIPTION, "delete"),
     SUBSCRIPTION_UPDATE(ResourceType.SUBSCRIPTION, "update"),
-    SUBSCRIPTION_SEEK(ResourceType.SUBSCRIPTION, "seek");
+    SUBSCRIPTION_SEEK(ResourceType.SUBSCRIPTION, "seek"),
+    SUBSCRIPTION_IAM_POLICY_SET(ResourceType.SUBSCRIPTION, "set_iam_policy"),
+    SUBSCRIPTION_IAM_POLICY_GET(ResourceType.SUBSCRIPTION, "get_iam_policy"),
+    SUBSCRIPTION_IAM_POLICY_DELETE(ResourceType.SUBSCRIPTION, "delete_iam_policy"),
+
+    IAM_POLICY_GET(ResourceType.IAM_POLICY, "get"),
+    IAM_POLICY_SET(ResourceType.IAM_POLICY, "set"),
+    IAM_POLICY_DELETE(ResourceType.IAM_POLICY, "delete");
```

Comment:
I am not sure of this being common for org/team/project. But its fine for time being, we can change it later if needed.

Reply:
org/team/project have hierarchal ordering while for the topic/sub/queue they occupy the same level in the resource hierarchy 

this is the reason why we can use common resource action for the former (can infer based on path) while for the latter we need more context since we cannot know what type the leaf id is.